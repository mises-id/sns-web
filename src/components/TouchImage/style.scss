.img-box{
  width:100vw;
  height:70vh;
  display: block;
  overflow: hidden;
  position: relative;
}
.img{
  width: 100%;
  position: absolute;
}