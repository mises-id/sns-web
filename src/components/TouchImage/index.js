/*
 * @Author: lmk
 * @Date: 2021-12-06 09:34:18
 * @LastEditTime: 2021-12-06 13:36:12
 * @LastEditors: lmk
 * @Description: 
 */
/* eslint-disable jsx-a11y/alt-text */
import React, { useEffect, useRef } from "react";
import PinchZoom from 'pinch-zoom-js'
import "./style.scss";
const TouchImage = (props) => {
  const ref = useRef(null)
  useEffect(() => {
    let el = document.querySelector('#img');
    new PinchZoom(el, {});
  }, [ref])
  return (
    <div className="img-box" >
      <img src={props.src} className="img" id="img"/>
    </div>
  );
};

export default TouchImage;
