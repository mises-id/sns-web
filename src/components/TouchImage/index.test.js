/*
 * @Author: lmk
 * @Date: 2021-07-15 10:12:21
 * @LastEditTime: 2021-12-06 09:34:34
 * @LastEditors: lmk
 * @Description: image test
 */
import React from 'react';
import {shallow} from 'enzyme';
import TouchImage from '.';
describe('TouchImage', () => {
  it('should render correctly', () => {
    const component = shallow(<TouchImage/>)
    expect(component).toMatchSnapshot();
  });
});