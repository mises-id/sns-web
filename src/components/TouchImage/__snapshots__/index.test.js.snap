// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`TouchImage should render correctly 1`] = `
<div
  className="img-box"
>
  <img
    className="img"
    id="img"
  />
</div>
`;
